landlab/components/flow_director/flow_direction_dinf.py

```diff
@@ -147,7 +147,7 @@ def flow_directions_dinf(grid,
     # with tarboton's 1997 algorithm, orthogonal link first, then diagonal.
     # has shape, (nnodes, 8 triangles, 2 neighbors)
     n_at_node = grid.neighbors_at_node
-    dn_at_node = grid._diagonal_neighbors_at_node
+    dn_at_node = grid.diagonal_adjacent_nodes_at_node
```

Comment:
@kbarnhart Perfect! Thank you! I actually meant to go over these exact issues when we talked today but forgot.

`adjacent` seems to be more standard in graph theory (Wikipedia has a lovely [glossary of terms](https://en.wikipedia.org/wiki/Glossary_of_graph_theory_terms)), which is why I started to use it for diagonals and d8s. I think we should also start to use "adjacent" instead of "neighbor" in this context for *links* as well. So "adjacent" would mean the node at the other end of a link-like element.

We still use the word "neighbor" in a more general manner for other stuff (like neighbor links and neighbor cells), which makes the term more confusing. If we stick with the word "neighbor", we really should add "node" after it so we know what element we're talking about. For example, does `neighbors_at_node` return ids of *nodes*? *cells*? *links*? The answer is *nodes* but I don't think it's clear from the name.

Something to also keep in mind is that users can use `RasterModelGrid` (or `HexModelGrid` or `RadialModelGrid`, etc.) and not have any idea about diagonals (or d8s) and that's the way it should be. diagonals are really just an add-on to a specific grid type. Thus, I think we should have (for example),
* `adjacent_nodes_at_node`
* `diagonal_adjacent_nodes_at_node`
* `d8_adjacent_nodes_at_node`

If we had `link_adjacent_node_at_nodes`, users who didn't know anything about diagonals, either because they weren't using them on `RasterModelGrid` or were using some other grid (`HexModelGrid`, say), would be confused with the extra modifier. Our general interface shouldn't change because of diagonals on one type of grid.

Reply:
I like consistency with graph theory terminology, and wish I'd paid more attention to it in originally naming our elements. But I worry that we might break users' code if we switch from "neighbor" to "adjacent". One solution is to make the change but keep (and deprecate) neighbor for a while. But I'm ok with either. That said, I think it would be inconsistent and confusing to have "adjacent" for diags and d8s and "neighbor" for links, because d8s include both links and diags. Better I think to use either "adjacent" or "neighbor" for all three. So I like Eric's proposed solution above. I also agree that we should avoid *link_adjacent_nodes_at_node*.

Reply:
My take would be to have an "plain" reference to neighbors/adjacents refer only to the true links, largely for backward compatibility but also to maintain consistency with the other grid types. Others would then be, e.g., `d8_neighbor_nodes_at_node`, `diagonal_neighbor_nodes_at_node`. Per Greg, definitely don't think we should mix "neighbor" and "adjacent", since in common english those would mean exactly the same thing.

If we're going to break code anyway by changing these names, the choice between `neighbor` and `adjacent` seems pretty unimportant. I guess I have a slight preference for `neighbor`, just for backwards familiarity.

Agree with Greg it's VERY important to retain what we have as a deprecated method, potentially for quite a while. You can stop it appearing in future documentation by removing the LLCATS.

Reply:
Thanks all!

I've now deprecated `neighbors_at_node` and `active_neighbors_at_node`. However, I haven't added `neighbor_nodes_at_node` or `active_neighbor_nodes_at_node` because it seemed weird to add these new methods and then immediately deprecate them.

I don't think we ever had anything like `diagonal_neighbor_nodes_at_node` or `active_diagonal_neighbor_nodes_at_node` as public methods (except `diagonal_cells`, which, oddly, was almost that) so I didn't create (and then deprecate) them. There was, however, a private method called `_diagonal_neighbors_at_node` but, because that was an internal data structure, so I figured it was fine to remove/rename it.